assets/helpers/paymentIntegrations/stripeCheckout.js

```diff
@@ -1,13 +1,147 @@
 // @flow
 
+/**
+ * This module contains a copy of pages/oneoff-contributions/helpers/ajax so that
+ * the new payment flow can use this code without any impact on the existing
+ * infrastructure.
+ *
+ * There are a couple of differences though:
+ * - the amount: instead of fixing it when the callback is created, it should
+ *   be provided by the `getState` function
+ * - the actions: what happens upon error/success is view-dependent, so the
+ *   actions are provided as parameters too. As a result, `dispatch` can
+ *   go away (which makes the module completely independant of React)
+ * - tracking conversions: this should be decoupled from the payment itself and
+ *   moved upstream, in the success handler
+ *
+ * The latter module can be removed entirely once the new payment flow becomes the one
+ * and only contribution endpoint.
+ */
+
+
+import { addQueryParamsToURL } from 'helpers/url';
+import {
+  derivePaymentApiAcquisitionData,
+  type ReferrerAcquisitionData,
+  type PaymentAPIAcquisitionData,
+} from 'helpers/tracking/acquisitions';
+import type { Participations } from 'helpers/abTests/abtest';
+import type { IsoCurrency } from 'helpers/internationalisation/currency';
+import type { OptimizeExperiments } from 'helpers/tracking/optimize';
+
+import * as cookie from 'helpers/cookie';
+import { logException } from 'helpers/logger';
 
 // ----- Setup ----- //
 
 let stripeHandler = null;
 
+const ONEOFF_CONTRIB_ENDPOINT = window.guardian.paymentApiStripeEndpoint;
+
+function stripeOneOffContributionEndpoint(testUser: ?string) {
+  if (testUser) {
+    return addQueryParamsToURL(
+      ONEOFF_CONTRIB_ENDPOINT,
+      { mode: 'test' },
+    );
+  }
+
+  return ONEOFF_CONTRIB_ENDPOINT;
+}
+
+// ----- Types ----- //
+
+type PaymentApiStripeExecutePaymentBody = {|
+  paymentData: {
+    currency: IsoCurrency,
+    amount: number,
+    token: string,
+    email: string
+  },
+  acquisitionData: PaymentAPIAcquisitionData,
+|};
 
 // ----- Functions ----- //
 
+function requestData(
+  abParticipations: Participations,
+  paymentToken: string,
+  currency: IsoCurrency,
+  referrerAcquisitionData: ReferrerAcquisitionData,
+  optimizeExperiments: OptimizeExperiments,
+  getData: Function,
+) {
+  const { user, amount } = getData();
```

Comment:
What is the advantage of this, as opposed to fixing the amounts in the callback?

Reply:
The callback is setup at rendering time, while it is called on submit. Since I am not using Redux state to store the form data, I need to read it off the form elements, which must happen on submit.

Reply:
I see. Thanks for explaining.